05.ls/ls_command.rb

```diff
@@ -8,23 +8,38 @@
 PERMISSION = { '0' => '---', '1' => '--x', '2' => '-w-', '3' => '-wx', '4' => 'r--', '5' => 'r-x', '6' => 'rw-', '7' => 'rwx' }.freeze
 
 def main
-  params = ARGV.getopts('l')
-  opt = OptionParser.new
-  path = opt.parse(ARGV)[0] || '.'
-  if params['l']
-    Dir.chdir(path) do
-      file_data_list = make_l_option_file
-      show_l_option(file_data_list)
+  params = parse_params
+  gathered_files = show_dot_file(params)
+  files = sorted_files(gathered_files, params)
```

Comment:
僕だったらこんな感じで書くかも。

```suggestion
  files = collect_files(params)
  sorted_files = sort_files(files, params)
```

以下、詳しい説明です。

たしかにgatherには「集める」という意味がありますが、なぜかプログラミングではあまり見かけない気がします。
collectとか、findとか、もしくはUnixコマンドにならってglobとか、そういった名前の方がポピュラーなイメージです。
今回はメソッド名の方で「collect files = ファイルを集める」と表現し、戻り値を受け取る変数はただのfilesとしました（collected filesのように書いてもいいですが、少し冗長な気もします）。

次に、ファイルを並び替える処理ですが、「並び替える = sort（動詞）」です。sortedは「並び替えられた」という形容詞的な意味を持つので、メソッドよりも変数名に付ける方が良いです。
というわけで、メソッド名をsort files（ファイルを並び替える）に、戻り値を受け取る変数をsorted files（並び替えられたファイル）としました。

以前お知らせしたかもしれませんが、このあたりの英語の使い分けは以下の記事で説明しているのであらためて読んでみてください〜（難しいですよね、英語……）。

[モデルやメソッドに名前を付けるときは英語の品詞に気をつけよう \- Qiita](https://qiita.com/jnchito/items/459d58ba652bf4763820)